src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -171,37 +165,37 @@ public final HostConnectionCounters getHostConnectionCounters() {
     return accessibleDatacenters;
   }
 
-  public Optional<JmxCredentials> getJmxCredentialsForCluster(String clusterName) {
-    Optional<JmxCredentials> jmxCreds = Optional.ofNullable(jmxAuth);
-    if (jmxCredentials != null && jmxCredentials.containsKey(clusterName)) {
-      jmxCreds = Optional.of(jmxCredentials.get(clusterName));
-    }
+  public JmxCredentials getJmxCredentialsForCluster(Node node) {
+    JmxCredentials nodeCredentials = node.getJmxCredentials().orElse(jmxAuth);
+
+    if (nodeCredentials == null && jmxCredentials != null) {
```

Comment:
if i got this right, this will preference jmx credentials in the following order:
1. backend stored in the backend
2. default credentials stored in the yaml
3. per-cluster credentials stored in the yaml

I would have thought that `per-cluster credentials, even if they are now deprecated, came second in the list?

(Can we also add unit tests for these types of preference checks between the various variant combinations)

Reply:
thanks for addressing this @denniskline 👍